Sources/SwiftProtobuf/Enum.swift

```diff
@@ -17,7 +17,7 @@
 
 import Swift
 
-public protocol Enum: RawRepresentable, Hashable, CustomDebugStringConvertible, FieldType, MapValueType {
+public protocol Enum: RawRepresentable, Hashable, CustomDebugStringConvertible {
     init()
     init?(jsonName: String)
     init?(protoName: String)
```

Comment:
> It does feel like we should be able to somehow reuse FieldNameMap here to provide name/number conversions.

This is a great idea. 👍  There may be some tricks we have to play with unrecognized cases, but I think we can make it work.